### ISSUE TITLE
use inject() w/ non-observer SFC

### ISSUE THREAD

**Issue by leader22** [mobxjs/mobx-react#277]
```js
const { inject, Provider } = mobxReact;

const SFC = ({ foo }) => <div>NG: {foo}</div>;
const Injected = inject('foo')(SFC);

ReactDOM.render(
  <Provider foo="foo">
    <Injected />
  </Provider>,
  document.getElementById('mount')
);
```

This code ends up with `Uncaught TypeError: Cannot read property 'render' of undefined`.
at https://github.com/mobxjs/mobx-react/blob/master/src/utils/utils.js#L2

Workarounds are
```js
// Works
class CC extends React.Component {
  render() {
    return <div>OK: {this.props.foo}</div>;
  }
}
const Injected = inject('foo')(CC);

// Also works
const SFC = ({ foo }) => (<div>OK: {foo}</div>);
const Injected = inject('foo')(observer(SFC));
```

Is this an expected behavior?

or code should be fixed like this?
```diff
export function isStateless(component) {
-  return !component.prototype.render;
+  return !(component.prototype && component.prototype.render);
}
```

pkg versions are

```
mobx: ^3.1.11,
mobx-react: ^4.2.1,
react: ^15.6.0,
react-dom: ^15.6.0
```

**Comment by mweststrate** [comment]
@leader22 that could be a bug in the check indeed, although I would expect the prototype of a SFC to be `Function`, or does that not apply to arrow functions? Did adjusting above code fix it for you? Feel free to submit a PR in that case

**Comment by mweststrate** [comment]
Interesting, that check indeed doesn't seem to work for not-down compiled arrow functions, so, confirmed:
```
➜  mobx-state-tree git:(localstate) node
> var x = () => 3
undefined
> x.prototype
undefined
> var y = function() {}
undefined
> y.prototype
y {}
```

**Comment by leader22** [comment]
@mweststrate 
Thank you for checking!
So, I submit a PR( #280 ) to fix this.

**Comment by farwayer** [comment]
nice catch!

**Comment by mweststrate** [comment]
This was released as 4.2.2. Forgot to close story.
